respa_admin/forms.py

```diff
@@ -374,14 +374,15 @@ def save(self, commit=True):
 
         if saved_form or self.forms:
             for form in self.forms:
-                period = form.save(commit=commit)
-                if period.template_src and (
-                    form.has_changed() or form.days.has_changed()
-                ):
-                    period.template_src = None
-                    period.save()
-                if hasattr(form, "days"):
-                    form.days.save(commit=commit)
+                if form not in self.deleted_forms:
```

Comment:
`Formset.deleted_forms` tracks the form instances flagged for deletion. We need to skip these here or the deleted Periods will be added back in again as new instances.